### DISCUSSION TITLE
Physics for skirts when character sits down - any way to be more "modest"?

### DISCUSSION THREAD

**Discussion by alankent** [pixiv/three-vrm#1464] [Help]
When adding a "sitting" animation clip to a model, the VRM spring bone for skirt is a little, err, revealing if you look from the front. Is there any way to overcome this? I like the skirt to kick out when standing, but when sitting...

![image](https://github.com/user-attachments/assets/e4db7230-dc2b-4c5d-9ba6-6c05a3d37032)

![image](https://github.com/user-attachments/assets/4bf95902-31ac-4e5c-9c06-48a124fa32a8)

**Comment by 0b5vr** [comment]
The skirt issue when sitting is a very common problem among VRM models, and there are no ways to resolve this on the app side since it's a model problem.

This usually comes from the mesh shape, the skinning weight, and the parameter/nature of the SpringBone system.
We are yet to find the best solution to this problem. However, the inside and plane colliders recently introduced in `VRMC_springBone_extended_collider` might make it look somewhat better.

**Reply by alankent** [reply]
Thank you. I was afraid that was the answer. When standing, I want the character to have the blown out skirt, as designed, so I don't want to permanently modify the model.

Do you think it makes sense to "animate" the springbone strengths at run time (so when playing a "sitting" animation clip, I also animate the spring bone strengths? E.g. increase gravity or reduce stiffness?

For now my solution is "frame the shot so you don't notice! ;-) https://youtu.be/dJ6LeKcnUvM

**Reply by 0b5vr** [reply]
> Do you think it makes sense to "animate" the springbone strengths at run time (so when playing a "sitting" animation clip, I also animate the spring bone strengths? E.g. increase gravity or reduce stiffness?

I'm not sure the SpringBone system in three-vrm supports that with ease but that might be a viable approach yes.

**Reply by alankent** [reply]
I did some initial digging, noting here in case anyone else comes across this thread. A challenge is gravity is stored per joint, so you have to find all the skirt bones (not the hair bones), then individually adjust the weights in each chain inside the custom VRM springbone JSON. I need to learn more about whether those properties can be captured in AnimationClip tracks.
